### PR TITLE
Fix smoketest failure at keyboard interruption

### LINKED ISSUE #3285: Gracefully handle JSON exception on alarm task
The stack trace bellow happened when I killed the smoketest locally, I guess geth was killed and the rpc call returned an empty string.

catching the json error and re-raising a `EthNodeCommunicationError` should be sufficient to gracefully handle this.

## Problem Definition

```
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
  File "/home/hack/work/brainbot/raiden/raiden/tasks.py", line 116, in _run
    self.loop_until_stop()
  File "/home/hack/work/brainbot/raiden/raiden/tasks.py", line 151, in loop_until_stop
    latest_block = self.chain.get_block(block_identifier='latest')
  File "/home/hack/work/brainbot/raiden/raiden/network/blockchain_service.py", line 53, in get_block
    return self.client.web3.eth.getBlock(block_identifier=block_identifier)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/eth.py", line 141, in getBlock
    [block_identifier, full_transactions],
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/manager.py", line 109, in request_blocking
    response = self._make_request(method, params)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/manager.py", line 92, in _make_request
    return request_func(method, params)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/cache.py", line 105, in middleware
    return make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/gas_price_strategy.py", line 18, in middleware
    return make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/attrdict.py", line 18, in middleware
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/formatting.py", line 48, in apply_formatters
    response = make_request(method, formatted_params)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/normalize_errors.py", line 9, in middleware
    result = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/middleware/exception_retry_request.py", line 80, in middleware
    return make_request(method, params)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/providers/rpc.py", line 70, in make_request
    response = self.decode_rpc_response(raw_response)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/providers/base.py", line 63, in decode_rpc_response
    return FriendlyJsonSerde().json_decode(text_response)
  File "/home/hack/work/brainbot/pyvenv3-dev/lib/python3.7/site-packages/web3/utils/encoding.py", line 260, in json_decode
    raise json.decoder.JSONDecodeError(err_msg, exc.doc, exc.pos)
json.decoder.JSONDecodeError: Could not decode '' because of Expecting value: line 1 column 1 (char 0).: line 1 column 1 (char 0)
2019-01-17T18:21:13Z <Greenlet "AlarmTask|Greenlet-2" at 0x6a2f2c2e7268: <bound method AlarmTask._run of <raiden.tasks.AlarmTask object at 0x6a2f2b86d5c0>>> failed with JSONDecodeError
```

### PR DESCRIPTION
**Pull request by LefterisJP**
Fix #3285

When attempting to stop the smoketest after the raiden client and geth
has started we could get a JSON decoding error or a keyboard interrupt
error and the test would be stuck.

This PR handles that.

### CODE REVIEW

File: raiden/tasks.py [hunk 1]
```diff
@@ -148,7 +150,11 @@ def loop_until_stop(self):
 
         sleep_time = self.sleep_time
         while self._stop_event.wait(sleep_time) is not True:
-            latest_block = self.chain.get_block(block_identifier='latest')
+            try:
+                latest_block = self.chain.get_block(block_identifier='latest')
+            except (KeyboardInterrupt, JSONDecodeError) as e:
```
Comment (andrevmatos) [inline raiden/tasks.py]: `KeyBoardInterrupt` doesn't look a lot like a `EthNodeCommunicationError`. Also, even if this is the main "waiting" point on which the `KeyboardInterrupt` could be raised inside the smoketest, I doubt it's the only one (gevent could be waiting on other greenlets maybe?). 
Maybe [here](https://github.com/raiden-network/raiden/blob/6782cc536c29f9bebbb9574fde43e425515c2795/raiden/ui/cli.py#L578-L598) could be a better place to handle the bubbling-up `KeyboardInterrupt`?

Comment (LefterisJP) [reply raiden/tasks.py]: @andrevmatos I moved it to the main function of the alarm task Runnable. The problem with what you are proposing is that the `KeyboardInterrupt` exception does not ever get out of the AlarmTask and just lets the smoke test hang. So it either needs to be handled inside the alarm task or somehow allowed to bubble out. Any suggestions?

Example trace:

```
[2/8] Starting Ethereum node
[3/8] Deploying Raiden contracts
[4/8] Setting up Raiden
[5/8] Starting Matrix transport
[6/8] Starting Raiden
^CTraceback (most recent call last):
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
  File "/home/lefteris/ew/raiden/raiden/tasks.py", line 118, in _run
    self.loop_until_stop()
  File "/home/lefteris/ew/raiden/raiden/tasks.py", line 154, in loop_until_stop
    latest_block = self.chain.get_block(block_identifier='latest')
  File "/home/lefteris/ew/raiden/raiden/network/blockchain_service.py", line 53, in get_block
    return self.client.web3.eth.getBlock(block_identifier=block_identifier)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/eth.py", line 141, in getBlock
    [block_identifier, full_transactions],
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/manager.py", line 109, in request_blocking
    response = self._make_request(method, params)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/manager.py", line 92, in _make_request
    return request_func(method, params)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/cache.py", line 105, in middleware
    return make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/gas_price_strategy.py", line 18, in middleware
    return make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/attrdict.py", line 18, in middleware
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/formatting.py", line 48, in apply_formatters
    response = make_request(method, formatted_params)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/normalize_errors.py", line 9, in middleware
    result = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/formatting.py", line 50, in apply_formatters
    response = make_request(method, params)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/middleware/exception_retry_request.py", line 80, in middleware
    return make_request(method, params)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/providers/rpc.py", line 55, in make_request
    **self.get_request_kwargs()
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/utils/request.py", line 25, in make_post_request
    session = _get_session(endpoint_uri)
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/utils/request.py", line 17, in _get_session
    cache_key = generate_cache_key((args, kwargs))
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/utils/caching.py", line 39, in generate_cache_key
    in value
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/utils/caching.py", line 26, in generate_cache_key
    return generate_cache_key(to_bytes(text=value))
  File "/home/lefteris/.virtualenvs/raiden/lib/python3.7/site-packages/web3/utils/caching.py", line 24, in generate_cache_key
    return hashlib.md5(value).hexdigest()
KeyboardInterrupt
2019-01-22T15:26:36Z <Greenlet "AlarmTask|Greenlet-2" at 0x7fe96ed5ae18: <bound method AlarmTask._run of <raiden.tasks.AlarmTask object at 0x7fe968b5b160>>> failed with KeyboardInterrupt

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 716, in gevent._greenlet.Greenlet.run
  File "/home/lefteris/ew/raiden/raiden/raiden_service.py", line 416, in _run
    self.stop_event.wait()
  File "src/gevent/event.py", line 240, in gevent._event.Event.wait
  File "src/gevent/event.py", line 140, in gevent._event._AbstractLinkable._wait
  File "src/gevent/event.py", line 117, in gevent._event._AbstractLinkable._wait_core
  File "src/gevent/event.py", line 119, in gevent._event._AbstractLinkable._wait_core
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 59, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/_greenlet_primitives.py", line 63, in gevent.__greenlet_primitives.SwitchOutGreenletWithLoop.switch
  File "src/gevent/__greenlet_primitives.pxd", line 35, in gevent.__greenlet_primitives._greenlet_switch
KeyboardInterrupt
2019-01-22T15:26:36Z <Greenlet "RaidenService|Greenlet-1" at 0x7fe96ed5ad08: <bound method RaidenService._run of <RaidenService 67a5e21e>>> failed with KeyboardInterrupt
```

Comment (LefterisJP) [reply raiden/tasks.py]: Ah thank you @andrevmatos for this insight. The smoketest was indeed not calling `runnable.get()` for the underlying RaidenService. This is why the KeyboardInterrupt was not bubbling up. Should be fixed with: https://github.com/raiden-network/raiden/pull/3300/commits/d700e0248d2592c506d34ff329e837389ed1bc09#diff-f9ae242ea5c3f431cbb17b9539236b09R592

File: raiden/ui/cli.py [hunk 2]
```diff
@@ -589,6 +589,7 @@ def _run_smoketest():
                 else:
                     success = True
             finally:
+                app.raiden.get()
```
Comment (andrevmatos) [inline raiden/ui/cli.py]: Ops.. `get` should be after `stop`, stop tells raiden to gracefully exit, `get` waits for it and re-raises any exception that may have been raised.

**Pull request merged**